Fails to build on Arch Linux due to ffmpeg breaking change

I'm guessing this project was designed for older ffmpeg versions and cmake tries to compile against new ones?
```
pancake@cute ~/tmp/obs-ios-camera-source $ make
[  5%] Built target libcnary
[ 32%] Built target libplist
[ 38%] Built target libusbmuxd
[ 80%] Built target libimobiledevice
[ 88%] Built target portal
[ 89%] Building C object CMakeFiles/obs-ios-camera-source.dir/src/ffmpeg-decode.c.o
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c: In function ‘ffmpeg_decode_init’:
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c:27:19: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   27 |     decode->codec = avcodec_find_decoder(id);
      |                   ^
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c: In function ‘ffmpeg_decode_free’:
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c:55:9: warning: ‘avcodec_close’ is deprecated [-Wdeprecated-declarations]
   55 |         avcodec_close(decode->decoder);
      |         ^~~~~~~~~~~~~
In file included from /home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.h:32,
                 from /home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c:18:
/usr/include/libavcodec/avcodec.h:2387:5: note: declared here
 2387 | int avcodec_close(AVCodecContext *avctx);
      |     ^~~~~~~~~~~~~
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c: In function ‘ffmpeg_decode_audio’:
/home/pancake/tmp/obs-ios-camera-source/src/ffmpeg-decode.c:209:52: error: ‘AVCodecContext’ has no member named ‘channels’
  209 |     convert_speaker_layout((uint8_t)decode->decoder->channels);
      |                                                    ^~
make[2]: *** [CMakeFiles/obs-ios-camera-source.dir/build.make:104: CMakeFiles/obs-ios-camera-source.dir/src/ffmpeg-decode.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:112: CMakeFiles/obs-ios-camera-source.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

Straight up removing the line in the error actually makes it build and work just fine on OBS 30.2.1, didn't test sound though

--- a/src/ffmpeg-decode.c
+++ b/src/ffmpeg-decode.c
@@ -52,8 +52,12 @@ void ffmpeg_decode_free(struct ffmpeg_decode *decode)
 {
     if (decode->decoder)
     {
+#if LIBAVCODEC_VERSION_MAJOR < 61
         avcodec_close(decode->decoder);
         av_free(decode->decoder);
+#else
+        avcodec_free_context(&decode->decoder);
+#endif
     }
 
     if (decode->frame)
@@ -206,7 +210,11 @@ bool ffmpeg_decode_audio(struct ffmpeg_decode *decode,
     audio->samples_per_sec = decode->frame->sample_rate;
     audio->format = convert_sample_format(decode->frame->format);
     audio->speakers =
+#if LIBAVCODEC_VERSION_MAJOR < 61
     convert_speaker_layout((uint8_t)decode->decoder->channels);
+#else
+    convert_speaker_layout((uint8_t)decode->decoder->ch_layout.nb_channels);
+#endif
 
     audio->frames = decode->frame->nb_samples;
 

--- a/src/ffmpeg-decode.h
+++ b/src/ffmpeg-decode.h
@@ -39,7 +39,11 @@ extern "C" {
 struct ffmpeg_decode
 {
 	AVCodecContext *decoder;
+#if LIBAVCODEC_VERSION_MAJOR < 59
 	AVCodec *codec;
+#else
+	const AVCodec *codec;
+#endif
 
 	AVFrame *frame;
 

